dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/NeuralNetwork.java

```diff
@@ -185,9 +185,14 @@ public void train(final INDArray input, final int label) {
   public INDArray[] backPropagateTo(final int end,
                                     final INDArray[] activations,
                                     final INDArray label) {
+    if (end == 0) {
+      throw new IllegalArgumentException("The ending layer cannot be the first layer: " +
```

Comment:
I'm not sure if this is correct. We don't have a separate 'input layer', do we? For example, the first layer in the sample conf file is a fully connected layer, which can have errors.